### PR TITLE
devx: add OWNERS files

### PR DESCRIPTION
**Pull request by jhchabran**
Following up on [RFC-551](https://docs.google.com/document/d/1TleM8nX7kJA9QySYj7IwVojzBf4A0KwzkKbQ8-dwnjQ/edit#heading=h.trqab8y0kufp), this PRs addes a few OWNERS files on our stuff.

### CODE REVIEW

File: dev/sg/OWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# See https://github.com/sourcegraph/codenotify for documentation.
+
+**/* @mrnugget @sourcegraph/dev-experience
```
Comment (jhchabran) [inline dev/sg/OWNERS:3]: @mrnugget heads up, added you in there :)

Comment (mrnugget) [reply dev/sg/OWNERS:3]: Nice :) Thanks!

**Review by bobheadxi — APPROVED** [review]
cool

File: dev/ci/OWNERS [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# See https://github.com/sourcegraph/codenotify for documentation.
+
+annotate.sh @sourcegraph/dev-experience
```
Comment (bobheadxi) [inline dev/ci/OWNERS:3]: Should this be in `dev/enterprise/ci/scripts` instead?

File: dev/ci/OWNERS [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+go-build.sh @sourcegraph/dev-experience
+go-test.sh @sourcegraph/dev-experience
+sentry-capture.sh @sourcegraph/dev-experience
```
Comment (bobheadxi) [inline dev/ci/OWNERS:6]: Likewise

File: dev/ci/OWNERS (hunk 2 above)
Comment (jhchabran) [reply dev/ci/OWNERS:3]: Yes indeed. I'll move it in a different PR.

**Pull request merged**